File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Get user input
    string s = get_string("Text: ");

    // Count number of letters
    int letters = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if ((islower(s[i])) || (isupper(s[i])))
        {
            letters++;
        }
    }

    // Initial count of a word
    int words = 0;
    if (s[0] != 32)
    {
        words++;
    }

    // Count of  words
    for (int i = 0; i < strlen(s); i++)
    {
        if (isspace(s[i]))
        {
            if (s[i + 1] != 32)
            {
                words++;
            }
        }
    }

    // Count number of sentences
    int sentences = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentences++;
        }
    }

    // Calculations
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    // Coleman - liau index
    float index = (0.0588 * L - 0.296 * S - 15.8);

    // Output
    if (round(index) >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (round(index) < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }
}

File: pset6/readability.py
from cs50 import get_string

text = get_string("Text: ")

lenght = len(text)
letter = []
for i in range(1, int(lenght)):
    letter.append(text)

# For  count of letters
letters = 0
for i in range(len(text)):
    if (text[i].isalpha()):
        letters += 1

# For initializing the count of words
words = 0
if (str.isalpha(text[0])):
    words += 1

# For  count of words
for i in range(len(text)):
    if (str.isspace(text[i]) and not str.isspace(text[i+1])):
        words += 1

# Count the number of sentences
sentences = 0
for i in range(len(text)):
    if (text[i] == '.' or text[i] == '!' or text[i] == "?"):
        sentences += 1
# Calculations
L = (letters) / (words) * 100
S = (sentences) / (words) * 100

# Coleman - liau index
index = (0.0588 * L - 0.296 * S - 15.8)

# Output
if (round(index) >= 16):
    print("Grade 16+")
elif (round(index) < 1):
    print("Before Grade 1")
else:
    r = int(round(index))
    print(f"Grade {r}")